xmtp/src/conversation.rs

```diff
@@ -40,17 +43,37 @@ impl<'c, A> SecretConversation<'c, A>
 where
     A: XmtpApiClient,
 {
-    pub fn new(
+    pub(crate) fn new(
         client: &'c Client<A>,
         peer_address: Address,
         // TODO: Add user's own contacts as well
         members: Vec<Contact>,
-    ) -> Self {
-        Self {
+    ) -> Result<Self, ConversationError> {
+        let obj = SecretConversation {
             client,
-            peer_address,
+            peer_address: peer_address.clone(),
             members,
-        }
+        };
+        obj.client.store.insert_or_ignore_user(StoredUser {
+            user_address: obj.peer_address(),
+            created_at: now(),
+            last_refreshed: 0,
+        })?;
+        obj.client
+            .store
+            .insert_or_ignore_conversation(StoredConversation {
+                peer_address: obj.peer_address(),
+                convo_id: obj.convo_id(),
+                created_at: now(),
+                convo_state: ConversationState::Uninitialized as i32,
+            })?;
+        Ok(obj)
+    }
+
+    pub fn convo_id(&self) -> String {
```

Comment:
We might want to move this into a utility function. I can see convo ids being created from places other than the conversation (to find conversations from an invite, for example).